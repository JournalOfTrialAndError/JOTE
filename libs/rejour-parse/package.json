{
  "name": "rejour-parse",
  "version": "0.0.1",
  "dependencies": {
    "unified": "^10.1.1",
    "unist-util-filter": "^4.0.0",
    "unist-util-map": "^3.0.0",
    "xast-util-from-xml": "^2.0.1"
  },
  "devDependencies": {
    "unist-util-remove-position": "^4.0.1"
  }
}
