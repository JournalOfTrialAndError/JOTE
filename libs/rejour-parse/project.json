{
  "root": "libs/rejour-parse",
  "sourceRoot": "libs/rejour-parse/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/rejour-parse/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/rejour-parse"],
      "options": {
        "jestConfig": "libs/rejour-parse/jest.config.js",
        "passWithNoTests": true
      }
    },
    "build": {
      "executor": "@nrwl/node:package",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/rejour-parse",
        "tsConfig": "libs/rejour-parse/tsconfig.lib.json",
        "packageJson": "libs/rejour-parse/package.json",
        "main": "libs/rejour-parse/src/index.ts",
        "assets": ["libs/rejour-parse/*.md"]
      }
    }
  },
  "tags": []
}
