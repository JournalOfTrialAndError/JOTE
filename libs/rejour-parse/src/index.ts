export * from './lib/rejour-parse'
