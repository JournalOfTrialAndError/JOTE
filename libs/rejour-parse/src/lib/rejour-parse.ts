import { fromXml } from 'xast-util-from-xml'

import { ParserFunction } from 'unified'
import { Root, Node } from 'xast'
import { filter } from 'unist-util-filter'
import { map } from 'unist-util-map'
import { Element } from 'xast-util-from-xml/lib'

export function rejourParse() {
  const parser: ParserFunction<Root> = (doc) => {
    // Assume options.
    // const settings = /** @type {Options} */ (this.data('settings'))

    const rawTree = fromXml(doc)
    const filteredTree = filter(rawTree, (node: any) => {
      return !(node.type === 'text' && node.value.replace(/\n +/, '') === '')
    })
    const tree = map(filteredTree!, (node: Node) => {
      if (node.type !== 'element') return node
      const element = node as Element

      return {
        type: 'element',
        tagName: element.name,
        properties: element.attributes,
        children: element.children,
        position: element.position,
      }
    })

    return tree as Root
  }

  Object.assign(this, { Parser: parser })
}
