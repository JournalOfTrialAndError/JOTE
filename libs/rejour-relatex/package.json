{
  "name": "rejour-relatex",
  "version": "0.0.1"
}
