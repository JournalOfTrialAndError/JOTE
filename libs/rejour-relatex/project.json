{
  "root": "libs/rejour-relatex",
  "sourceRoot": "libs/rejour-relatex/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/rejour-relatex/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/rejour-relatex"],
      "options": {
        "jestConfig": "libs/rejour-relatex/jest.config.js",
        "passWithNoTests": true
      }
    },
    "build": {
      "executor": "@nrwl/node:package",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/rejour-relatex",
        "tsConfig": "libs/rejour-relatex/tsconfig.lib.json",
        "packageJson": "libs/rejour-relatex/package.json",
        "main": "libs/rejour-relatex/src/index.ts",
        "assets": ["libs/rejour-relatex/*.md"]
      }
    }
  },
  "tags": []
}
