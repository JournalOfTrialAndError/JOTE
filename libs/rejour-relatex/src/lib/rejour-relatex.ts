import { toLtast, Options } from 'jast-util-to-ltast'
import { Root as JastRoot } from 'rejour'
import { Root as LtastRoot } from 'relatex'
import {
  Plugin,
  Processor as UnifiedProcessor,
  TransformCallback,
  Transformer,
} from 'unified'
import { VFile } from 'vfile'
type Processor = UnifiedProcessor<any, any, any, any>

/**
 * Bridge-mode.
 * Runs the destination with the new mdast tree.
 *
 */
function bridge(
  destination: Processor,
  options?: Options
): void | Transformer<JastRoot, JastRoot> {
  return (node, file, next) => {
    destination.run(toLtast(node, options), file, (error) => {
      next(error)
    })
  }
}

/**
 * Mutate-mode.
 * Further transformers run on the mdast tree.
 */
function mutate(
  options: void | Options | undefined = {}
): Transformer<JastRoot, JastRoot> | void {
  return (node) => {
    const result = toLtast(node, options)
    return result
  }
}

/**
 * Plugin to bridge or mutate to rehype.
 *
 * If a destination is given, runs the destination with the new mdast
 * tree (bridge-mode).
 * Without destination, returns the mdast tree: further plugins run on that
 * tree (mutate-mode).
 *
 * @param destination
 *   Optional unified processor.
 * @param options
 *   Options passed to `jast-util-to-ltast`.
 */
const rejourRelatex = function (
  destination?: Processor | Options,
  options?: Options
) {
  let settings: Options | undefined
  let processor: Processor | undefined

  if (typeof destination === 'function') {
    processor = destination
    settings = options
  } else {
    settings = destination
  }

  if (settings?.document === undefined || settings.document === null) {
    settings = Object.assign({}, settings, { document: true })
  }

  return processor ? bridge(processor, settings) : mutate(settings)
} as Plugin<[Processor, Options?], JastRoot> &
  Plugin<[Options?] | void[], JastRoot, LtastRoot>

export default rejourRelatex
