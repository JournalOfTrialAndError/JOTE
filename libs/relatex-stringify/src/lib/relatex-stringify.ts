export function relatexStringify(): string {
  return 'relatex-stringify'
}
