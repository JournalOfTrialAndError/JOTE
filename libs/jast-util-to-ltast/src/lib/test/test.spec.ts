import { rejourParse } from 'rejour-parse'
import { rejourRelatex } from 'rejour-relatex'
import { relatexStringify } from 'relatex-stringify'
import { unified } from 'unified'

describe('fixtures', () => {
  const rejour = unified()
    .use(rejourParse)
    .use(rejourRelatex)
    .use(relatexStringify)
})
