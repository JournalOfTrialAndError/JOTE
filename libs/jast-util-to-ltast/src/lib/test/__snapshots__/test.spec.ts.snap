// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses correctly for article should match snapshot 1`] = `
"\\\\begin{document}

  

\\\\end{document}"
`;

exports[`parses correctly for complete should match snapshot 1`] = `
"version=\\"1.0\\" encoding=\\"utf-8\\"Classical Conditioning for Pain: The Development of a Customized Single-Case Experimental Design \\\\begin{document}

  

\\\\end{document}"
`;

exports[`parses correctly for environment should match snapshot 1`] = `
"\\\\begin{document}

  

\\\\end{document}"
`;

exports[`parses correctly for italic should match snapshot 1`] = `"\\\\emph{Hey}"`;

exports[`parses correctly for root should match snapshot 1`] = `
"\\\\begin{document}

  

\\\\end{document}"
`;

exports[`parses correctly for title should match snapshot 1`] = `"\\\\title{A fancy title\\\\n\\\\n}"`;
